### PR TITLE
fix(hadron-build, compass): restore Isolated and Readonly special behavior in packaged application COMPASS-8129

### PR DESCRIPTION
**Pull request by gribnoysup**
When we separated Compass compilation process from the app packaging, we stopped passing correct environment to the compile script, which lead to webpack config not setting environment correctly when bundling the assets. This turned both Compass isolated and readonly into "normal" editions even though the installer and the binary were still branded correctly.

This patch fixes the issue by being more explicit in where and how we set the expected env variable, I also modified the `Target` class, the main source of this information right now, that is used during build and compilation to not have an unsafe default anymore and throw if `HADRON_DISTRIBUTION` variable was not provided, this should ensure we don't accidentally break it anymore

### CODE REVIEW

File: packages/hadron-build/commands/verify.js [hunk 1]
```diff
@@ -1,53 +0,0 @@
...
-const run = require('./../lib/run');
-const runAsync = promisify(run);
-
-exports.command = 'verify [options]';
-exports.describe = 'Verify the current environment meets the app\'s requirements.';
-
-exports.builder = {
-  nodejs_version: {
-    describe: 'What version of node.js is required for this app?',
-    default: '^7.4.0'
```
Comment (gribnoysup) [inline packages/hadron-build/commands/verify.js:21]: Drive-by, just caught my eye how out of date this was, so I removed this script too

File: .evergreen/functions.yml [hunk 2]
```diff
@@ -405,7 +406,7 @@ functions:
           <<: *compass-env
           DEBUG: ${debug}
           npm_config_loglevel: ${npm_loglevel}
-          COMPASS_DISTRIBUTION: ${compass_distribution}
+          HADRON_DISTRIBUTION: ${compass_distribution}
```
Comment (gribnoysup) [inline .evergreen/functions.yml:409]: Changed it back to HADRON_ because otherwise it's hard to search through the code for all occurences, I'm totally up for changing this to COMPASS_ at some point, but we should probably do this for all the mentions of hadron

**Review by lerouxb — APPROVED** [review]

**Review by himanshusinghs — APPROVED** [review]
